packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -336,6 +336,14 @@ def test_metrics_containers_nan(dcos_api_session):
         assert metric_value == '', 'unexpected metric value: {}'.format(metric_value)
 
 
+@retrying.retry(wait_fixed=(2 * 1000), stop_max_delay=(150 * 1000))
+def test_zookeeper_metrics_collected(dcos_api_session):
+    for master in dcos_api_session.masters:
+        response = dcos_api_session.get('/metrics', scheme='http', host=master, port=61091)
```

Comment:
Why `port=61091`? Is this an internal interface? Does this assume that the test runner is executed on a master node? It's been a while since I worked with the api session interface, but `host=master` is confusing to me. `master` is not a valid DNS name.

Reply:
Good point! We could use the `/system/v1/metrics` route on AR, which means we don't need the scheme or port.

However, `master` is a variable here. What makes you think it is a DNS name?

Reply:
dcos-telegraf is outputting to `0.0.0.0:61091` which means the `/metrics` endpoint can be read from anywhere. I guess that's not ideal but currently I think we do not care about shielding metrics from the outside. `dcos_api_sessions.masters` returns the list of IP addresses specified the environment variable `MASTER_HOSTS` that we set before starting our CI. Since the test intends to assert that the metrics are reported on every master node we cannot utilize `/system/v1/metrics` through Admin Router. TBH I don't even know whether this is the same metrics stack that this endpoint exposes. I did not manage to use it in my test cluster via Admin Router. So I believe apart from the `retry` the test is exactly how it should be.

Reply:
> What makes you think it is a DNS name?

My brain made me think this, and clearly there was something wrong with my brain :)

Reply:
> dcos-telegraf is outputting to 0.0.0.0:61091 which means the /metrics endpoint can be read from anywhere. I guess that's not ideal but currently I think we do not care about shielding metrics from the outside.

Even more important than security is clear, rational thinking about what is a public interface and what not. We should always focus on testing a public interface first. It's what must "never" break. 

> TBH I don't even know whether this is the same metrics stack that this endpoint exposes.

That worries me a little bit. Can we invest a little more time, discussion, research in order to understand what's meant to be consumed by whom here? Ideally we know exactly what _external_ tooling expects, and test things from the point of view of this external tooling.

Reply:
Agreed, however I think that this is a larger topic that should be part of discussion with @leehambley

Reply:
> That worries me a little bit. Can we invest a little more time, discussion, research in order to understand what's meant to be consumed by whom here?

This is the Prometheus output which is enabled by default in Telegraf which exposes all metrics known to Telegraf in Promethus standard format.

- https://github.com/dcos/dcos/blob/279685203d9159526408f759cf6a5a700b25f7bc/gen/dcos-config.yaml#L1357-L1359

@jgehrcke if you have a suggestion to change the listener from `:61091` to something more concrete (`::1:61091` or whatever) we're all ears. (/cc @philipnrmn) 

This is the same endpoint you should be able to reach "from anywhere" and is our recommended public interface:

- https://docs.mesosphere.com/1.12/metrics/prometheus/